src/audio/buffer.c

```diff
@@ -76,6 +80,71 @@ struct comp_buffer *buffer_alloc(uint32_t size, uint32_t caps, uint32_t flags, u
 	return buffer;
 }
 
+#if CONFIG_ZEPHYR_DP_SCHEDULER
+int buffer_create_shadow_dp_queue(struct comp_buffer *buffer, bool at_input)
+{
+	if (buffer->stream.dp_queue_sink || buffer->stream.dp_queue_source) {
+		tr_err(&buffer_tr, "Only one shadow dp_queue may be attached to a buffer");
+		return -EINVAL;
+	}
+
+	struct dp_queue *dp_queue =
+		dp_queue_create(source_get_min_available(&buffer->stream._source_api),
+				sink_get_min_free_space(&buffer->stream._sink_api),
+				buffer->is_shared ? DP_QUEUE_MODE_SHARED : DP_QUEUE_MODE_LOCAL,
+				buf_get_id(buffer), &buffer->stream.runtime_stream_params);
+
+	if (!dp_queue)
```

Comment:
It may be good to log this, since we depend on data coming from topology/drivers to construct the queue.

Reply:
good idea, currently only creation of dp_queue is in log:
``<inf> dp_queue: dp_queue_create: DpQueue created, id: 0 shared: 1 min_available: 384 min_free_space 512, size 1024``